text/label/sentencepiece/apply.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import shutil
+import subprocess as sp
+import tempfile
+import os
+
+from sisyphus import Job, Task, tk, gs
+
+import i6_core.util as util
+
+try:
+    import sentencepiece
+except ImportError:
+    if not hasattr(gs, "WARNING_NO_SENTENCEPIECE") or gs.WARNING_NO_SENTENCEPIECE is True:
+        logging.warning(
+            "The package 'sentencepiece' is not installed in the manager python env. Please make sure it is installed "
+            "in the python environment running the Sisyphus worker. To suppress this warning set "
+            "'WARNING_NO_SENTENCEPIECE=False' in the settings.py"
+        )
+
+
+class ApplySentencepieceToTextJob(Job):
+    """
+    Apply sentencepiece model on a text file, basically a wrapper for spm.encode
+    """
+
+    def __init__(
+        self,
+        *,
+        text_file: tk.Path,
+        sentencepiece_model: tk.Path,
+        map_unk: bool = False,
+        gzip_output: bool = True,
+        mini_task: bool = False,
+    ):
+        """
+        :param text_file: words text file to convert to sentencepiece
+        :param sentencepiece_model: path to the trained sentencepiece model
+        :param map_unk: when encoding string to string, spm won't map oov symbol to <unk> but keep it as is.
+            This option forces the oov labels to be <unk> by cecking encoded indices.
+        :param gzip_output: use gzip on the output text
+        :param mini_task: if the Job should run locally, e.g. only a small (<1M lines) text should be processed
+        """
+        self.text_file = text_file
+        self.sentencepiece_model = sentencepiece_model
+        self.map_unk = map_unk
+
+        self.out_sentencepiece_text = self.output_path(
+            "words_to_sentencepiece.txt.gz" if gzip_output else "words_to_sentencepiece.txt"
+        )
+
+        self.mini_task = mini_task
+        self.rqmt = {"cpu": 1, "mem": 2, "time": 2}
+
+    def tasks(self):
+        if self.mini_task:
+            yield Task("run", mini_task=True)
+        else:
+            yield Task("run", rqmt=self.rqmt)
```

Comment:
we sometimes use 
```suggestion
        self.rqmt = {"cpu": 1, "mem": 2, "time": 2}

    def tasks(self):
        yield Task("run", rqmt=self.rqmt, mini_task=self.rqmt is None)
```
this way we could drop another parameter and the custom hash function

Reply:
Also fine with me. But you should declare the type of `self.rqmt: Optional[Dict[str, Any]]` or so, i.e. make it clear that this can be set to None.